packages/Search/test/DTK_BoostRTreeHelpers.hpp

```diff
@@ -0,0 +1,115 @@
+// objects into the tree.  The values of the maximum and the minimum number
+// of elements could be adjusted in principle but I didn't bother setting
+// this type as a template argument.
+using Parameter = boost::geometry::index::linear<16>;
+
+template <typename Value, typename Index>
+struct PairMaker
+{
+    template <typename T>
+    inline std::pair<Value, Index> operator()( T const &v ) const
+    {
+        return std::make_pair( v.value(), v.index() );
+    }
+};
+
+template <typename Indexable>
+using RTree =
+    boost::geometry::index::rtree<std::pair<Indexable, int>, Parameter>;
+
+// NOTE: The template parameter Indexable is used as an alias here
+template <typename View, typename Indexable = typename View::value_type>
+static RTree<Indexable> makeRTree( View const &objects )
+{
+    return RTree<Indexable>(
+        objects | boost::adaptors::indexed() |
+        boost::adaptors::transformed( PairMaker<Indexable, int>() ) );
+}
+
+// NOTE: The trailing return type required with C++11 :(
+template <typename Value>
+struct UnaryPredicate
+{
+    using Function = std::function<bool( Value const & )>;
+    UnaryPredicate( Function pred )
+        : _pred( pred )
+    {
+    }
+    inline bool operator()( Value const &val ) const { return _pred( val ); }
+    static UnaryPredicate makeAlwaysFalse()
+    {
+        return UnaryPredicate( []( Value const & ) { return false; } );
+    }
+    Function _pred;
+};
+
+template <typename Value>
+static auto translate(
+    DataTransferKit::Details::Intersects<DataTransferKit::Sphere> const &query )
+    -> decltype( boost::geometry::index::intersects( DataTransferKit::Box() ) &&
+                 boost::geometry::index::satisfies(
+                     UnaryPredicate<Value>::makeAlwaysFalse() ) )
+{
+    auto const sphere = query._geometry;
+    auto const radius = sphere.radius();
+    auto const centroid = sphere.centroid();
+    DataTransferKit::Box box;
+    DataTransferKit::Details::expand( box, sphere );
+    return boost::geometry::index::intersects( box ) &&
+           boost::geometry::index::satisfies(
+               UnaryPredicate<Value>( [centroid, radius]( Value const &val ) {
+                   boost::geometry::index::indexable<Value> indexableGetter;
+                   auto const &geometry = indexableGetter( val );
+                   return boost::geometry::distance( centroid, geometry ) <=
+                          radius;
+               } ) );
+}
+
+template <typename Value, typename Geometry>
+static auto
+translate( DataTransferKit::Details::Nearest<Geometry> const &query )
+    -> decltype( boost::geometry::index::nearest( Geometry(), 0 ) )
+{
+    auto const geometry = query._geometry;
+    auto const k = query._k;
+    return boost::geometry::index::nearest( geometry, k );
+}
+
+template <typename Indexable, typename InputView,
+          typename OutputView = Kokkos::View<int *, Kokkos::HostSpace>>
```

Comment:
Does that work with `OutputView` on the GPU and UVM?

Reply:
`InputView` memory space must be accessible from host (static assertion in `DTK_BoostRangeAdapters.hpp`)

I did not anticipated that `OutputView` would be a view on the device and I fill the indices directly from the host in `performQueries()`.  That does not "have to be" this way.

I guess it is me again abusing templates and using `OutputView` as an alias so that I can use it both when I spell out the return type of the function and when I declare the view later on.

Reply:
@Rombur want me to change anything there?

Reply:
No if it doesn't have to be on the host space then that's fine